src/fdb5/api/FDB.cc

```diff
@@ -123,6 +123,7 @@ void FDB::archive(const Key& key, const void* data, size_t length) {
 
     timer.stop();
     stats_.addArchive(length, timer);
+    return keyInternal;
```

Comment:
I don't think that this is correct. The registry has been attached here, but this key is still the pre-canonicalisation key.

You need to return from internal_->archive()

Reply:
I agree this is not the canonicalised key. I'll look into this again, but I think it might not be possible to return the fully canonicalised key due to it only being created in the recursive traversal of the schema.